x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/related_integrations/related_integrations.cy.ts

```diff
@@ -5,239 +5,407 @@
+    { package: 'system', version: '1.17.0', installed: true, enabled: true },
+  ];
+  const PREBUILT_RULE = createRuleAssetSavedObject({
+    name: PREBUILT_RULE_NAME,
+    index: [DATA_STREAM_NAME],
+    query: '*:*',
+    rule_id: 'rule_1',
+    related_integrations: RULE_RELATED_INTEGRATIONS.map((x) => omit(x, ['installed', 'enabled'])),
   });
 
-  context('integrations not installed', () => {
-    const rule = {
-      name: 'Related integrations rule',
-      integrations: ['Aws Cloudfront', 'Aws Cloudtrail', 'Aws Unknown', 'System'],
-      enabledIntegrations: '0',
-    };
-
-    before(() => {
-      cleanFleet();
-    });
-
-    beforeEach(() => {
-      login();
-      visit(DETECTIONS_RULE_MANAGEMENT_URL);
-      waitForRulesTableToShow();
-    });
+  beforeEach(() => {
+    login();
+    cleanFleet();
+    deleteAlertsAndRules();
+    addAndInstallPrebuiltRules([PREBUILT_RULE]);
+  });
 
-    it('should display a badge with the installed integrations on the rule management page', () => {
-      cy.get(INTEGRATIONS_POPOVER).should(
-        'have.text',
-        `${rule.enabledIntegrations}/${rule.integrations.length} integrations`
-      );
-    });
+  describe('integrations not installed', () => {
+    describe('rules management table', () => {
+      beforeEach(() => {
+        visitSecurityDetectionRulesPage();
+        disableAutoRefresh();
+      });
 
-    it('should display a popover when clicking the badge with the installed integrations on the rule management page', () => {
-      openIntegrationsPopover();
+      it('should display a badge with the installed integrations', () => {
+        cy.get(INTEGRATIONS_POPOVER).should(
+          'have.text',
+          `0/${RULE_RELATED_INTEGRATIONS.length} integrations`
+        );
+      });
 
-      cy.get(INTEGRATIONS_POPOVER_TITLE).should(
-        'have.text',
-        `[${rule.integrations.length}] Related integrations available`
-      );
-      cy.get(INTEGRATIONS).should('have.length', rule.integrations.length);
-      cy.get(INTEGRATIONS_STATUS).should('have.length', rule.integrations.length);
+      it('should display a popover when clicking the badge with the installed integrations', () => {
+        openIntegrationsPopover();
 
-      rule.integrations.forEach((integration, index) => {
-        cy.get(INTEGRATIONS).eq(index).should('contain', integration);
-        cy.get(INTEGRATIONS_STATUS).eq(index).should('have.text', 'Not installed');
+        cy.get(INTEGRATIONS_POPOVER_TITLE).should(
+          'have.text',
+          `[${RULE_RELATED_INTEGRATIONS.length}] Related integrations available`
+        );
+        cy.get(INTEGRATION_LINK).should('have.length', RULE_RELATED_INTEGRATIONS.length);
+        cy.get(INTEGRATION_STATUS).should('have.length', RULE_RELATED_INTEGRATIONS.length);
+
+        RULE_RELATED_INTEGRATIONS.forEach((integration, index) => {
+          cy.get(INTEGRATION_LINK).eq(index).contains(getIntegrationName(integration), {
```

Comment:
Wondering about the use of `eq(i)` here.

Can we guarantee that the order of the integrations links is always the same as defined in `RULE_RELATED_INTEGRATIONS`? Might it be a cause of flakiness in the future?

Reply:
As you can see prebuilt rule asset and tests reuse the same constant `RULE_RELATED_INTEGRATIONS`. Related integrations rendered in the popover are rendered based on `rule.related_integrations` array. This means the order is stable now and we catch any changes here.